src/serac/physics/tests/serac_thermal_structural_solver.cpp

```diff
@@ -140,6 +140,278 @@ TEST(dynamic_solver, dyn_solve)
+    ts_solver_secondary.setVelocity(*velo);
+    ts_solver_secondary.setTemperature(*temp);
+    ts_solver_secondary.setCouplingScheme(serac::CouplingScheme::OperatorSplit);
+
+    auto temp_gf_coef_secondary = ts_solver_secondary.temperature().gridFuncCoef();
+    auto visc_coef_secondary    = std::make_unique<mfem_ext::TransformedScalarCoefficient<mfem::Coefficient>>(
+        [offset, scale](double& x) -> double { return scale * x + offset; }, temp_gf_coef_secondary);
+    ts_solver_secondary.setViscosity(std::move(visc_coef_secondary));
+
+    ts_solver_secondary.initializeOutput(serac::OutputType::SidreVisIt, "");
+
+    // Construct the internal dynamic solver data structures
+    ts_solver_secondary.completeSetup();
+
+    // Ouput the initial state
+    ts_solver_secondary.outputState();
+
+    double t_intermediate_primary = 3.0;
+    // Perform time-integration
+    // (looping over the time iterations, ti, with a time-step dt).
+    bool last_step = false;
+    for (int ti = 1; !last_step; ti++) {
+      double dt_real = std::min(dt, t_intermediate_primary - t_primary);
+      t_primary += dt_real;
+      last_step = (t_primary >= t_intermediate_primary - 1e-8 * dt);
+
+      ts_solver_primary.advanceTimestep(dt_real);
+      primary_reload_cycle = ti;
+    }
+
+    // Output the final state
+    ts_solver_primary.outputState();
+
+    double t_intermediate_secondary = 2.0;
+    // Perform time-integration
+    // (looping over the time iterations, ti, with a time-step dt).
+    last_step = false;
+    for (int ti = 1; !last_step; ti++) {
+      double dt_real = std::min(dt, t_intermediate_secondary - t_secondary);
+      t_secondary += dt_real;
+      last_step = (t_secondary >= t_intermediate_secondary - 1e-8 * dt);
+
+      ts_solver_secondary.advanceTimestep(dt_real);
+      secondary_reload_cycle = ti;
+    }
+
+    // Output the final state
+    ts_solver_secondary.outputState();
+  }
+
+  MPI_Barrier(MPI_COMM_WORLD);
+
+  // Then restart the simulation, running each to a total of six cycles
+  {
+    // Create DataStore
+    axom::sidre::DataStore datastore;
+    serac::StateManager::initialize(datastore, "dynamic_solve_restart");
+    serac::StateManager::load(primary_reload_cycle, primary_datacoll_name);
+    serac::StateManager::load(secondary_reload_cycle, secondary_datacoll_name);
+
+    // What's awkward about using a mesh as the param is that we have to then extract the mesh
+    // instead of using the string ID we just used in the call to load()
+    // Note that just the 'auto' will not compile but only because ParMesh move ctor is explicitly deleted
+    auto& primary_mesh   = serac::StateManager::mesh(primary_datacoll_name);
+    auto& secondary_mesh = serac::StateManager::mesh(secondary_datacoll_name);
+
+    // initialize the dynamic solver object
+    ThermalSolid ts_solver_primary(1, therm_options, default_dynamic, "coupled_primary", &primary_mesh);
+    ts_solver_primary.setDisplacementBCs(ess_bdr, deform);
```

Comment:
Outside the scope of this PR but pretty sure we agreed to remove set and get from our function calls.

Reply:
Yeah, I think this is a holdover from before that decision.

Reply:
#611